starboard/common/test_main.cc

```diff
@@ -27,12 +26,19 @@ int InitAndRunAllTests(int argc, char** argv) {
 }
 }  // namespace
 
-// When we are building Evergreen we need to export SbEventHandle so that the
-// ELF loader can find and invoke it.
+#if defined(OS_STARBOARD)
+// For the Starboard OS define SbEventHandle as the entry point
 SB_EXPORT STARBOARD_WRAP_SIMPLE_MAIN(InitAndRunAllTests);
+
+#if !SB_IS(EVERGREEN)
+// Define main for non-Evergreen Starboard OS.
+int main(int argc, char** argv) {
+  SbRunStarboardMain(argc, argv, SbEventHandle);
+}
+#endif  // !SB_IS(EVERGREEN)
 #else
+// If the OS is not Starbaord use the regular main
```

Comment:
This path would be taken if `OS_STARBOARD` is not defined - but would this file ever be built, considering this folder is  called `//starboard` ? IOW do we need this #else ? (If we're not sure I'd propose leaving a `#error This combination should not happen` or some such.

Reply:
Here is where the water gets muddy. This case is actually used by ATV and could be used by other platforms in the future if we use the native OS as the platform but still use Starboard as a library e.g. SbPlayer.

Added a comment to explain that ATV uses it.